### PR TITLE
CMOB 1814 / Details Screen

### PR DESCRIPTION
**Pull request by DavitTsiskarashvili**
https://spaceneobank.atlassian.net/browse/CMOB-1814

Attention: More than half of the changed files are due to reformat of code

In this PR:
- Details screen shows every quiz the user has taken with scores
- Reformatted all code
- Changed app icon
- Changed regex into Georgian usernames
- Fixed GPA saving issue while re-login
- Fixed some design bugs

### CODE REVIEW

**Review by Gio481 — APPROVED** [review]

File: app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt [hunk 1]
```diff
@@ -4,12 +4,10 @@ package com.space.quiz_app.common.utils
 /**
 Regular expression pattern for validating usernames.
 The pattern requires the username to meet the following criteria:
-At least one lowercase letter
-At least one uppercase letter
-At least one digit
-Minimum length of 8 characters
+Minimum length of 2 characters
+ Username should be in Georgian
```
Comment (Gio481) [inline app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt]: format unda

File: app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt [hunk 2]
```diff
@@ -4,12 +4,10 @@ package com.space.quiz_app.common.utils
...
  */
 
 object QuizRegex {
```
Comment (Gio481) [inline app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt]: Quiz tu UserName

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizQuestionsDao.kt [hunk 3]
```diff
@@ -13,5 +13,4 @@ interface QuizQuestionsDao {
 
     @Query("SELECT * FROM question WHERE subjectTitle = :subjectTitle")
     suspend fun getQuestions(subjectTitle: String): List<QuizQuestionEntity>
```
Comment (Gio481) [inline app/src/main/java/com/space/quiz_app/data/local/dao/QuizQuestionsDao.kt]: getQuestionsBySubject

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt [hunk 4]
```diff
@@ -17,6 +17,9 @@ interface QuizUserDao {
     @Query("SELECT * FROM User_table WHERE isLoggedIn = true")
     suspend fun getUsernameIfLoggedIn(): QuizUserEntity?
 
+    @Query("UPDATE User_table SET isLoggedIn =:activeStatus WHERE username =:username")
+    suspend fun updateUserStatus(username: String, activeStatus: Boolean)
+
     @Query("SELECT * FROM User_table WHERE username=:username")
     suspend fun getUser(username: String): QuizUserEntity
```
Comment (Gio481) [inline app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt]: getUser gadaarqvi

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt [hunk 5]
```diff
@@ -17,6 +17,9 @@ interface QuizUserDao {
     @Query("SELECT * FROM User_table WHERE isLoggedIn = true")
     suspend fun getUsernameIfLoggedIn(): QuizUserEntity?
 
+    @Query("UPDATE User_table SET isLoggedIn =:activeStatus WHERE username =:username")
+    suspend fun updateUserStatus(username: String, activeStatus: Boolean)
```
Comment (Gio481) [inline app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt]: updateUserStatus შესაბამისი სახელი მოუფიქრე

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserSubjectsDao.kt [hunk 6]
```diff
@@ -20,5 +20,4 @@ interface QuizUserSubjectsDao {
 
     @Query("SELECT * FROM user_subject WHERE username=:username and quizTitle=:quizTitle")
     suspend fun getUserSpecificSubject(username: String, quizTitle: String): QuizUserSubjectEntity?
```
Comment (Gio481) [inline app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserSubjectsDao.kt]: getUserSpecificSubject რა specific subject ? :d

**Review by Gio481 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt (hunk 1 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt]: done

File: app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt (hunk 2 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/space/quiz_app/common/utils/QuizRegex.kt]: done

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizQuestionsDao.kt (hunk 3 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/space/quiz_app/data/local/dao/QuizQuestionsDao.kt]: done

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt (hunk 4 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt]: done

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt (hunk 5 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserDao.kt]: done

File: app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserSubjectsDao.kt (hunk 6 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/space/quiz_app/data/local/dao/QuizUserSubjectsDao.kt]: done

**Review by giorgimurgulia-space — APPROVED** [review]

**Review by bchmsl — APPROVED** [review]

**Review by AnaRusadze — APPROVED** [review]

**Pull request merged**